javascript-bank/bank.js

```diff
@@ -0,0 +1,55 @@
+/* exported Bank */
+
+function Bank() {
+  this.nextAccountNumber = 1;
+  this.accounts = [];
+  this.bank = { nextAccountNumber: this.nextAccountNumber, accounts: this.accounts };
+}
+
+Bank.prototype.openAccount = function (holder, balance) {
+  this.balance = balance;
+  this.holder = holder;
+  this.newAccount = [];
+  var returnNumber = this.nextAccountNumber;
+  if (Number.isInteger(this.balance)) {
+    if (this.balance > 0) {
+      this.newAccount = new Account(this.nextAccountNumber, this.holder);
+      this.newAccount.deposit(this.balance);
+      this.accounts.push(this.newAccount);
+      this.nextAccountNumber++;
+      return returnNumber;
+    }
+    return null;
+  } else {
+    return null;
+  }
+};
+
+Bank.prototype.getAccount = function (number) {
+
+  this.number = number;
```

Comment:
Please do not add additional properties not requested in the instructions. You can make local variables to the function to get the same effect without polluting the object the data is being stored in.